lib/src/form/core/formx.dart

```diff
@@ -0,0 +1,93 @@
+import 'package:equatable/equatable.dart';
+import 'package:flutter_formx/src/form/core/formx_field.dart';
+import 'package:flutter_formx/src/form/core/formx_state.dart';
+
+/// Vanilla implementation of [FormX].
+/// The state management implementations should hold one instance of this class.
+///
+/// [FormX] is the core class to handle forms. [T] stands for the type used to
+/// identify the fields such as an enum or a string to later access each of its
+/// fields.
```

Comment:
```suggestion
```suggestion
/// [FormX] is the core class to handle forms. [T] stands for the type of the key
/// used to identify the fields such as an enum or a string to later access each
/// of its fields.
```
```